lms/djangoapps/grades/models.py

```diff
@@ -534,6 +534,21 @@ def _emit_grade_calculated_event(grade):
     def _cache_key(cls, course_id):
         return f"subsection_grades_cache.{course_id}"
 
+    @classmethod
+    def clear_grade(cls, user_id, course_key):
```

Comment:
I'd suggest making these names a bit more descriptive, here maybe something like `delete_subsection_grades_for_learner`.

Also, I have a suggestion for how this should interact with the subsection override model, but I went into a lot of detail in that comment.